app/models/foreman_azure_rm/azure_rm_compute.rb

```diff
@@ -119,6 +122,27 @@ def ip_addresses
       []
     end
 
+    def data_disks
+      return nil unless @azure_vm.storage_profile.data_disks
+      disks = @azure_vm.storage_profile.data_disks
+    end
```

Comment:
It would be easier to rewrite the method to return an empty array : 

``` ruby
    def data_disks
      @data_disks ||= @azure_vm.storage_profile.data_disks || []
    end
```